src/System.IO.Pipelines.Extensions/ReadableBufferSequence.cs

```diff
@@ -16,41 +16,33 @@ public ReadableBufferSequence(ReadableBuffer buffer) : this()
 
         public bool TryGet(ref Position position, out ReadOnlyMemory<byte> item, bool advance = true)
         {
-            if (position == Position.First)
+            if (position == default)
             {
                 // First is already sliced
                 item = _buffer.First;
                 if (advance)
                 {
                     if (_buffer.Start.IsEnd)
                     {
-                        position = Position.AfterLast;
+                        position = Position.Infinity;
                     }
                     else
                     {
-                        position.ObjectPosition = _buffer.Start.Segment.Next;
-                        if (position.ObjectPosition == null)
-                        {
-                            position = Position.AfterLast;
-                        }
+                        position.Set(_buffer.Start.Segment.Next);
                     }
                 }
                 return true;
             }
-            else if (position == Position.AfterLast)
+            else if (position == Position.Infinity)
```

Comment:
If you made `Position.Infinity` why did you removed `First`?

Reply:
First is default. Infinity is a special value.

Reply:
`DateTime.MinValue` is default too, but it exists because it's a well known value. Also such constants allow you to have uniform code style. Just compare:
```csharp
if (position = default) { }
else if (position = Position.Ininity) { }
// and
if (position = Position.First) { }
else if (position = Position.Ininity) { }
```

Reply:
Yeah, there is a trade off here. I used to have Position.First for the reason you stated, but then the code was not consistent. In some places people were using First and in some default. Then on the other side people have to worry if First == default or if they need to do two checks. Removal of First makes it clear.